### PR TITLE
Debug data structures, dumping their content to files or stdout. 

### PR DESCRIPTION
**Pull request by hamax97**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,8 @@ TEST_DIR            = test
 CC                  = gcc
 CXX                 = g++
 CFLAGS              =
-CXXFLAGS            =
+#CXXFLAGS	    =
+CXXFLAGS            = -DDEBUG_ITERATION_NUM=20 -DDEBUG_PARTICLE_NUM=0
```
Comment (BalmungSan) [inline Makefile]: I would prefer this to do not be hard coded in the Makefile, but rather that it is passed as an argument to `make`.

**Review by BalmungSan — COMMENTED** [review]
:+1:
Just the changes we already discussed.

File: src/cpp/debug.cpp [hunk 2]
```diff
@@ -15,129 +16,115 @@ extern Vector *accelerations;
...
-               "", tangent_forces_file);
-  write_header(step, particle_num, "Forces.",
-               "x_component y_component", forces_file);
-  write_header(step, particle_num, "Accelerations.",
-               "x_component y_component", accelerations_file);
-  write_header(step, particle_num, "Velocities.",
-               "x_component y_component", velocities_file);
-  write_header(step, particle_num, "Displacements.",
-               "x_component y_component", displacements_file);
+  int hola = contacts_size;
```
Comment (BalmungSan) [inline src/cpp/debug.cpp:26]: Please rename this variable.

More than that. why it is even there? It is not used.

**Review by BalmungSan — APPROVED** [review]

File: src/cpp/debug.cpp (hunk 2 above)
Comment (hamax97) [reply src/cpp/debug.cpp:26]: Sorry, I was temporarily fixing a compilation error due that _contacts\_size_ was not used

**Pull request merged**